src/lang-promql/complete.ts

```diff
@@ -0,0 +1,117 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2020 The Prometheus Authors
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import { AutocompleteContext, Completion, CompletionResult } from "@codemirror/next/autocomplete";
+import { LSPBody, LSPClient } from "./lsp/client";
+import { CompletionItem, TextEdit } from "vscode-languageserver-types";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function isIterable(obj: any): boolean {
+  return obj != null && typeof obj[ Symbol.iterator ] === 'function';
```

Comment:
How about using only the strict comparison operators, `!==` instead of `!=`, otherwise `null` also matched `undefined` (and if that's intended, then it's better to make it explicit IMO).

Reply:
yes that's intended, I was hesitated by just do it like this:

```typescript
return obj && typeof obj[ Symbol.iterator ] === 'function'
```

but on stackoverflow they all check if it's null instead of doing like the way I wanted. So I let it like that.